src/diffusers/textual_inversion_utils.py

```diff
@@ -0,0 +1,101 @@
+import torch
+
+
+class TextualInversionMixin:
+    r"""
+    Mixin class for adding textual inversion tokens and embeddings to the tokenizer and text encoder with method:
+    - [`~TextualInversionMixin.load_textual_inversion_embeddings`]
+    - [`~TextualInversionMixin.add_textual_inversion_embedding`]
+
+    Class attributes:
+    - **textual_inversion_tokens** (`List[str]`): list of tokens added to the tokenizer's vocabulary and the text
+      encoder's embedding matrix
+    """
+    textual_inversion_tokens = []
+
+    def load_textual_inversion_embeddings(self, embeddings):
+        r"""
+        Loads textual inversion embeddings.
+
+        Receives a dictionary with the following keys:
+        - `token`: name of the token to be added to the tokenizers' vocabulary
+        - `embedding`: path to the embedding of the token to be added to the text encoder's embedding matrix
+
+        Alternatively, it can receive a list of pathes to embedding dictionaries, where the keys are the tokens and the
+        values are the embeddings. In that case, it will iterate through the list and add the tokens and embeddings to
+        the tokenizer's vocabulary and the text encoder's embedding matrix.
+
+        Iters through the dictionary and adds the token to the tokenizer's vocabulary and the embedding to the text
+        encoder's embedding matrix.
+        """
+
+        if isinstance(embeddings, dict):
+            for token, embedding_path in embeddings.items():
+                # check if token in tokenizer vocab
+                # if yes, raise exception
+                if token in self.tokenizer.get_vocab():
+                    raise ValueError(
+                        f"Token {token} already in tokenizer vocabulary. Please choose a different token name."
+                    )
+
+                embedding_dict = torch.load(embedding_path)
```

Comment:
uff that's a bit of double loading here. The embeddings should already be the fully loaded state dict . Why do we need to "double" load here? Is this the AUTO1111 format?

Reply:
We have a case for passing a list and another a "token": "file" format, and for both cases, diffusers and auto1111. Does that make sense?

Reply:
@patrickvonplaten Embeddings are only ever loaded once. The previous argument name was misleading as it wasn't actually the embedding. I renamed the argument so that it's more explicit albeit more verbose.